src/react_native/utils.cljs

```diff
@@ -37,3 +40,24 @@
         props       (when (map? first-child) first-child)
         children    (if props (rest argv) argv)]
     [props children]))
+
+(defn convert-prop-value
+  "Based on `reagent.impl.template/kv-conv`.
+  Takes the prop map of a reagent component and returns a React-valid property JS Object
+  by transforming kebab-case keys -> camelCase, maps -> JS Objects and vectors -> arrays.
+
+  This version adds support to recursively transform properties inside vectors, to have a
+  more consistent developer experience in React Native."
+  [x]
+  (cond
+    (reagent.util/js-val? x) x
+    (reagent.util/named? x)  (name x)
+    (map? x)                 (reduce-kv template/kv-conv #js {} x)
+    (vector? x)              (transforms/map-array convert-prop-value x)
+    (coll? x)                (clj->js x)
+    (ifn? x)                 (fn [& args]
+                               (apply x args))
+    :else                    (clj->js x)))
+
+;; We override the default reagent implementation with the one that supports vectors.
+(set! template/convert-prop-value convert-prop-value)
```

Comment:
please move it to `status-im.core` under `;;;; re-frame RN setup`, we shouldn't override anything in the utils ns itself